projects/ngx-maplibre-gl/src/lib/source/geojson/feature.component.ts

```diff
@@ -1,53 +1,53 @@
 import {
   Component,
   forwardRef,
-  Inject,
-  Input,
   OnDestroy,
   OnInit,
   ChangeDetectionStrategy,
+  input,
+  model,
+  inject,
 } from '@angular/core';
 import { GeoJSONSourceComponent } from './geojson-source.component';
 
 /**
  * `mgl-feature` - a feature component
  * [ngx] inside {@link GeoJSONSourceComponent} only
- * 
+ *
  * @category Source Components
  */
 @Component({
-    selector: 'mgl-feature',
-    template: '',
-    changeDetection: ChangeDetectionStrategy.OnPush,
-    standalone: true,
+  selector: 'mgl-feature',
+  template: '',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
 })
-export class FeatureComponent
-  implements OnInit, OnDestroy, GeoJSON.Feature<GeoJSON.GeometryObject> {
+// GeoJSON.Feature<GeoJSON.GeometryObject>
+export class FeatureComponent implements OnInit, OnDestroy {
+  /** Init injection */
+  private readonly geoJSONSourceComponent = inject<GeoJSONSourceComponent>(
+    forwardRef(() => GeoJSONSourceComponent)
+  );
   /** Init input */
-  @Input() id?: number; // FIXME number only for now https://github.com/mapbox/mapbox-gl-js/issues/2716
-  /** Init input */
-  @Input() geometry: GeoJSON.GeometryObject;
-  /** Init input */
-  @Input() properties: any;
-  type: 'Feature' = 'Feature';
+  readonly id = model<number>(); // FIXME number only for now https://github.com/mapbox/mapbox-gl-js/issues/2716
+  readonly geometry = input.required<GeoJSON.GeometryObject>();
+  readonly properties = input<any>();
 
-  private feature: GeoJSON.Feature<GeoJSON.GeometryObject>;
+  readonly type: 'Feature' = 'Feature';
```

Comment:
I'm not sure this is interesting in terms of keeping as a property, I would hard code it in the relevant places.